packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -158,29 +183,67 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
       wire.emit(PaidStreamingExtensionEvents.REQUEST, peerAccount);
     });
 
+    wire.paidStreamingExtension.once(PaidStreamingExtensionEvents.PSE_HANDSHAKE, async () => {
+      const peerAccount = wire.paidStreamingExtension && wire.paidStreamingExtension.peerAccount;
+      const pseAccount = wire.paidStreamingExtension && wire.paidStreamingExtension.pseAccount;
+
+      if (!this.amSeeder) {
```

Comment:
Is there a better way to determine if I should create the channel?

Reply:
Been testing what would be the best way of handling this.
I think that on
https://github.com/statechannels/monorepo/blob/89202f6ae7e8bb8d96e0a41c771f446930b62874/packages/web3torrent/src/library/web3torrent-lib.ts#L115
when we set the pseAccount, we could pass other variables, including something like "_seederMode_". That way, each peer will let the other know what role is having (**seeder** or **leecher**)
Though, as I commented before, I think this doesn't consider the duplex nature of wires, nor the role changing nature of peers.